### PR TITLE
[WIP] Fixes for julia 0.7 deprecations

### PR DESCRIPTION
**Pull request by tlnagy**
It looks like upstream packages are finally building on the nightlies so it's time to get Compose working as well.

### CODE REVIEW

**Comment by tlnagy** [comment]
We can still run femtocleaner on this PR, right?. There's still plenty of cruft from previous julia version bumps that it can clean up.

**Comment by bjarthur** [comment]
yes, but, it would caught everything you did so far.  no need to put manual sweat into it.  let's wait to merge this until after we tag a release, just right before dataframes merges.

File: src/container.jl [hunk 1]
```diff
@@ -4,18 +4,18 @@
...
 
 # The basic Container which defines a coordinate transform for its children.
-type Context <: Container
+mutable struct Context <: Container
     # Bounding box relative to the parent's coordinates
     box::BoundingBox
 
     # Context coordinates used for children
-    units::Nullable{UnitBox}
+    @compat units::Union{UnitBox, Nothing}
```
Comment (bjarthur) [inline src/container.jl]: i had to [revert](https://github.com/GiovineItalia/Compose.jl/pull/281/commits/066d8f1d7805ea100befc6e2c7f55908369f293c) this and similar changes as they causes tests to fail.  was this for 0.7?

Comment (tlnagy) [reply src/container.jl]: This change was needed for v0.7, but I hadn't verified that my solution worked on v0.6. See https://github.com/JuliaLang/julia/pull/23642

**Pull request merged**